listenbrainz/background/listens_importer/maloja.py

```diff
@@ -0,0 +1,96 @@
+import json
+from datetime import datetime, timezone
+from typing import Any, Iterator
+
+from flask import current_app
+from more_itertools import chunked
+
+from listenbrainz.background.listens_importer.base import BaseListensImporter
+
+
+class MalojaListensImporter(BaseListensImporter):
+    """Importer for Maloja listening history exports."""
+
+    def process_import_file(self, import_task: dict[str, Any]) -> Iterator[list[dict[str, Any]]]:
+        """Read the Maloja JSON export and yield filtered scrobbles in batches."""
+        from_date = import_task["from_date"]
+        to_date = import_task["to_date"]
+
+        with open(import_task["file_path"], mode="r", encoding="utf-8") as infile:
+            data = json.load(infile)
+
+        scrobbles = data.get("scrobbles", [])
+        filtered_scrobbles: list[dict[str, Any]] = []
+        for item in scrobbles:
+            try:
+                timestamp = datetime.fromtimestamp(item["time"], tz=timezone.utc)
+            except (KeyError, TypeError, ValueError, OSError):
+                current_app.logger.error("Invalid Maloja timestamp in item: %s", item, exc_info=True)
+                continue
+
+            if from_date <= timestamp <= to_date:
+                filtered_scrobbles.append(item)
+
+        if not filtered_scrobbles:
+            return iter(())
+
+        return chunked(filtered_scrobbles, self.batch_size)
+
+    def parse_listen_batch(self, batch: list[dict[str, Any]]) -> list[dict[str, Any]]:
+        """Convert Maloja scrobbles into ListenBrainz listen payloads."""
+        listens: list[dict[str, Any]] = []
+
+        for item in batch:
+            try:
+                track = item.get("track", {}) or {}
+                artists = track.get("artists", []) or []
+                if not artists:
+                    continue
+
+                track_name = track.get("title")
+                if not track_name:
+                    continue
+
+                artist_name = ", ".join(artists)
+                track_metadata: dict[str, Any] = {
+                    "artist_name": artist_name,
+                    "track_name": track_name,
+                }
+
+                album = track.get("album") or {}
+                album_title = album.get("albumtitle")
+                if album_title:
+                    track_metadata["release_name"] = album_title
+
+                additional_info: dict[str, Any] = {
+                    "submission_client": self.importer_name,
+                    "music_service": "maloja",
+                }
+
+                album_artists = album.get("artists", []) or []
+                if album_artists and album_artists != artists:
+                    additional_info["release_artist_name"] = ", ".join(album_artists)
+
+                duration_seconds = item.get("duration")
```

Comment:
This should instead use track.length if present, according to the [Maloja API docs](https://github.com/krateng/maloja/blob/9e44cc3ce6d4259c32026ba50ee934e024b43a7a/API.md): 
> duration:	How long the track was played for in seconds

> track > length: The full length of the track in seconds

We don't have a formalized field for "duration played", but if you want to ensure it is in there so we don't lose that data, I would suggest something like `track_metadata.additional_info.duration_played` in seconds, or `track_metadata.additional_info.ms_played` in milliseconds, whichever sounds clearest to you.